examples/pytorch/argo/ogb_example.py

```diff
@@ -0,0 +1,301 @@
+            )
+            if args.save_pred:
+                np.savetxt(
+                    args.save_pred + "%02d" % epoch,
+                    pred.argmax(1).cpu().numpy(),
+                    "%d",
+                )
+            print("Eval Acc {:.4f}".format(eval_acc))
+            if eval_acc > best_eval_acc:
+                best_eval_acc = eval_acc
+                best_test_acc = test_acc
+            print(
+                "Best Eval Acc {:.4f} Test Acc {:.4f}".format(
+                    best_eval_acc, best_test_acc
+                )
+            )
+
+    print("Avg epoch time: {}".format(avg / (epoch - 4)))
+    return best_test_acc
+
+
+if __name__ == "__main__":
+    argparser = argparse.ArgumentParser("multi-gpu training")
+    argparser.add_argument(
+        "--gpu",
+        type=int,
+        default=0,
+        help="GPU device ID. Use -1 for CPU training",
+    )
+    argparser.add_argument("--num-epochs", type=int, default=20)
+    argparser.add_argument("--num-hidden", type=int, default=256)
+    argparser.add_argument("--num-layers", type=int, default=3)
+    argparser.add_argument("--fan-out", type=str, default="5,10,15")
+    argparser.add_argument("--batch-size", type=int, default=1000)
+    argparser.add_argument("--val-batch-size", type=int, default=10000)
+    argparser.add_argument("--log-every", type=int, default=20)
+    argparser.add_argument("--eval-every", type=int, default=1)
+    argparser.add_argument("--lr", type=float, default=0.003)
+    argparser.add_argument("--dropout", type=float, default=0.5)
+    argparser.add_argument("--dataset", type=str, default="ogbn-products", choices= ["ogbn-papers100M", "ogbn-products"])
+    argparser.add_argument(
+        "--num-workers",
+        type=int,
+        default=4,
+        help="Number of sampling processes. Use 0 for no extra process.",
+    )
+    argparser.add_argument("--save-pred", type=str, default="")
+    argparser.add_argument("--wd", type=float, default=0)
+    args = argparser.parse_args()
+
+    device = th.device("cpu")
+
+    # load ogbn-products data
+    data = DglNodePropPredDataset(args.dataset)
+    splitted_idx = data.get_idx_split()
+    train_idx, val_idx, test_idx = (
+        splitted_idx["train"],
+        splitted_idx["valid"],
+        splitted_idx["test"],
+    )
+    graph, labels = data[0]
+    nfeat = graph.ndata.pop("feat").to(device)
+    labels = labels[:, 0].to(device)
+
+    in_feats = nfeat.shape[1]
+    n_classes = (labels.max() + 1).item()
+    # Create csr/coo/csc formats before launching sampling processes
+    # This avoids creating certain formats in each data loader process, which saves momory and CPU.
+    graph.create_formats_()
+    # Pack data
+    data = (
+        train_idx,
+        val_idx,
+        test_idx,
+        in_feats,
+        labels,
+        n_classes,
+        nfeat,
+        graph,
+    )
+
+    # Run 10 times
+    test_accs = []
+    for i in range(10):
```

Comment:
I think we should run it only once as we do it  only once on _ARGO version.

Reply:
Ok, updated.